File: mini-commerce/src/main/java/me/hero/minicommerce/item/application/DeleteItemService.java
package me.hero.minicommerce.item.application;

import lombok.RequiredArgsConstructor;
import me.hero.minicommerce.item.application.port.in.DeleteItemUseCase;
import me.hero.minicommerce.item.application.port.out.DeleteItemPort;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class DeleteItemService implements DeleteItemUseCase {

  private final DeleteItemPort deleteItemPort;

  @Override
  public void deleteItem(long itemId) {
    deleteItemPort.deleteItem(itemId);
  }
}

File: mini-commerce/src/main/kotlin/me/hero/minicommerce/item/application/DeleteItemService.kt
package me.hero.minicommerce.item.application

import me.hero.minicommerce.item.application.port.`in`.DeleteItemUseCase
import me.hero.minicommerce.item.application.port.out.DeleteItemPort
import org.springframework.stereotype.Service

@Service
class DeleteItemService (
    private val deleteItemPort: DeleteItemPort,
) : DeleteItemUseCase{

    override fun deleteItem(itemId: Long) {
        deleteItemPort.deleteItem(itemId)
    }
}